server/meta.jsx

```diff
@@ -37,9 +40,22 @@ function validatePaymentsSDKUrl({ pathname, query, hash }) {
             throw new TypeError(`Unexpected non-string key for sdk url: ${ key }`);
         }
 
-        if (!val.match(/^[a-zA-Z0-9_,-@.]+$/) && !val.match(/^\*$/)) {
+        if (!val.match(/^[a-zA-Z0-9+_,-@.]+$/) && !val.match(/^\*$/)) {
             throw new Error(`Unexpected characters in query key for sdk url: ${ key }=${ val }`);
         }
+
+        if (key === SDK_QUERY_KEYS.MERCHANT_ID) {
+            const merchantValues = val.split(",");
+            merchantValues.forEach(merchantValue => {
+                if (!emailMaxLengthRegex.test(merchantValue)) {
```

Comment:
is there a reason to use a regex here? Wouldn't this be simpler (and I assume more performant):

```suggestion
                if (merchantValue.length > 256) {
```

Though I assume you'd pull out the `256` into a `const` at the top like `const maxEmailLength = 256` or something.

Though, also, this suggests that the max email address length is 320: https://www.rfc-editor.org/errata/eid1690#:~:text=It%20should%20say%3A-,In%20addition%20to%20restrictions%20on%20syntax%2C%20there%20is%20a%20length,total%20length%20of%20320%20characters.